src/java.base/share/classes/sun/nio/fs/NativeBuffers.java

```diff
@@ -88,6 +88,8 @@ static NativeBuffer getNativeBufferFromCache(int size) {
      * Returns a native buffer, of at least the given size. The native buffer
      * is taken from the thread local cache if possible; otherwise it is
      * allocated from the heap.
+     * <p>
+     * The returned NativeBuffer is guaranteed not to be asynchronously closeable.
```

Comment:
This class, and the temporary buffer cache in sun.nio.ch.Util, are intended to be used with try-finally. There isn't any notion of  asynchronously close so maybe it would best to not add this comment to these sources.

Reply:
That is clear to me but I am trying to prevent future redundant guarding. Anyway, I will remove the comments.

Reply:
> That is clear to me but I am trying to prevent future redundant guarding. Anyway, I will remove the comments.

The faraway use sites look much better now. The performance sensitive usages need close attention. Can you summarise the changes to DatagramChannel, are you creating a NoOpScopeAcquisition for every  send/receive? This is low level code where we do do something different to avoid the try-with-resources.